### PR TITLE
Add MSBuild task that will get the hash out from an assembly

### PR DESCRIPTION
**Pull request by joperezr**
PTAL: @weshaggard 

This adds a task that we will use in core-setup in order to get the commit hash out from an assembly.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GetVersionFromAssembly : Task
+    {
+        [Required]
+        public string PathToAssembly { get; set; }
+
+        [Output]
+        public string AssemblyVersion { get; set; }
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs]: This shouldn't be AssemblyVersion it should be AssemblySHA or something.

File: src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                if (match.Success && match.Groups.Count > 1)
+                {
+                    AssemblyVersion = match.Groups[1].Value;
+                }
+                else
+                {
+                    Log.LogError($"{PathToAssembly} has an invalid ProductVersion or does not contain a commit hash.");
+                    return false;
+                }
+            } catch (Exception)
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs]: new line needed after brace.

File: src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            {
+                Log.LogError($"{PathToAssembly} does not exist.");
+                return false;
+            }
+            try
+            {
+                Regex regex = new Regex(@"Commit Hash: ([A-Za-z0-9].*)");
+                string productVersion = FileVersionInfo.GetVersionInfo(PathToAssembly).ProductVersion;
+                Match match = regex.Match(productVersion);
+                if (match.Success && match.Groups.Count > 1)
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs]: @ericstj should we go ahead and have this task expose the other version information from this assembly?

File: src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs (hunk 1 above)
Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs]: good point, will fix.

File: src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs (hunk 2 above)
Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs]: Will add.

**Comment by joperezr** [comment]
Closing this since it doesn't make sense anymore since we will get this info on a different way.

**Pull request closed without merging**